src/opm/output/eclipse/AggregateNetworkData.cpp

```diff
@@ -0,0 +1,752 @@
+        denw += ndp.ndDenw*ndp.ndWpr;
+        deng += ndp.ndDeng*ndp.ndGpr;
+    }
+    deno = (opr > 0.) ? deno/opr : 0.;
+    denw = (wpr > 0.) ? denw/wpr : 0.;
+    deng = (gpr > 0.) ? deng/gpr : 0.;
+
+    return {deno, denw, deng, opr, wpr, gpr};
+}
+
+namespace INode {
+std::size_t entriesPerNode(const std::vector<int>& inteHead)
+{
+    return inteHead[Opm::RestartIO::Helpers::VectorItems::NINODE];
+}
+
+Opm::RestartIO::Helpers::WindowedArray<int>
+allocate(const std::vector<int>& inteHead)
+{
+    using WV = Opm::RestartIO::Helpers::WindowedArray<int>;
+
+    return WV {
+        WV::NumWindows{ nodmax(inteHead) },
+        WV::WindowSize{ entriesPerNode(inteHead) }
+    };
+}
+
+int numberOfBranchesConnToNode(const Opm::Schedule& sched, const std::string& nodeName, const size_t lookup_step)
+{
+    auto& network = sched[lookup_step].network();
+    if (network.has_node(nodeName)) {
+        int noBranches = network.downtree_branches(nodeName).size();
+        noBranches = (network.uptree_branch(nodeName).has_value()) ? noBranches+1 : noBranches;
+        return noBranches;
+    } else {
+        std::stringstream str;
+        str << "actual node: " << nodeName << " has not been defined at report time: " << lookup_step+1;
+        throw std::invalid_argument(str.str());
+    }
+}
+
+int cumNumberOfBranchesConnToNode(const Opm::Schedule& sched, const std::string& nodeName, const size_t lookup_step)
+{
+    auto& network = sched[lookup_step].network();
+    std::size_t ind_name = 0;
+    int cumNoBranches = 1;
+    auto result = findInVector<std::string>(network.insert_index_nd_names(), nodeName);
+    if (result.first) {
+        ind_name = result.second;
+        if (ind_name == 0) {
+            return cumNoBranches;
+        } else {
+            for (std::size_t n_ind = 0; n_ind < ind_name; n_ind++) {
+                cumNoBranches += numberOfBranchesConnToNode(sched,  network.insert_index_nd_names()[n_ind], lookup_step);
+            }
+            return cumNoBranches;
+        }
+    } else {
+        std::stringstream str;
+        str << "actual node: " << nodeName << " has not been defined at report time: " << lookup_step+1;
+        throw std::invalid_argument(str.str());
+    }
+}
+
+template <class INodeArray>
+void staticContrib(const Opm::Schedule&     sched,
+                   const std::string&       nodeName,
+                   const std::size_t        lookup_step,
+                   INodeArray&              iNode)
+{
+    //
+    using Ix = VI::INode::index;
+    iNode[Ix::NoBranchesConnToNode] = numberOfBranchesConnToNode(sched, nodeName, lookup_step);
+    iNode[Ix::CumNoBranchesConnToNode] = cumNumberOfBranchesConnToNode(sched, nodeName, lookup_step);
+    if (sched.hasGroup(nodeName, lookup_step)) {
+        iNode[Ix::Group] = sched.getGroup(nodeName, lookup_step).insert_index();
+    }
+    iNode[Ix::FixedPresNode] = (fixedPressureNode(sched, nodeName, lookup_step)) ? 1 : 0;
+    iNode[4] = 1;
```

Comment:
What is `4`?

Reply:
The meaning of the value of this index is unknown, but it has to be 1!

Reply:
Ok, add a comment?

Reply:
OK - I will write comments for these special cases.